prow/plugins/trigger/generic-comment_test.go

```diff
@@ -612,34 +617,30 @@ func TestHandleIssueComment(t *testing.T) {
 			t.Fatalf("failed to set presubmits: %v", err)
 		}
 
-		var pr *struct{}
-		if tc.IsPR {
-			pr = &struct{}{}
-		}
-		event := github.IssueCommentEvent{
-			Action: github.IssueCommentActionCreated,
+		event := github.GenericCommentEvent{
+			Action: github.GenericCommentActionCreated,
 			Repo: github.Repo{
 				Owner:    github.User{Login: "org"},
 				Name:     "repo",
 				FullName: "org/repo",
 			},
-			Comment: github.IssueComment{
-				Body: tc.Body,
-				User: github.User{Login: tc.Author},
-			},
-			Issue: github.Issue{
-				User:        github.User{Login: tc.PRAuthor},
-				PullRequest: pr,
-				State:       tc.State,
-			},
-		}
-		if len(tc.IssueLabels) > 0 {
-			event.Issue.Labels = tc.IssueLabels
+			Body:        tc.Body,
+			User:        github.User{Login: tc.Author},
+			IssueAuthor: github.User{Login: tc.PRAuthor},
+			IssueState:  tc.State,
+			IsPR:        tc.IsPR,
 		}
 
-		if err := handleIC(c, nil, event); err != nil {
+		// In some cases handleGenericComment can be called twice for the same event.
+		// For instance on Issue/PR creation and modification.
+		// Let's call it twice to ensure idempotency.
+		if err := handleGenericComment(c, nil, event); err != nil {
+			t.Fatalf("Didn't expect error: %s", err)
+		}
+		if err := handleGenericComment(c, nil, event); err != nil {
```

Comment:
If we run the handler twice for every test case we may cover up errors that occur if the handler is only run once (the normal case). The easiest way to catch this kind of bug in the unit tests would be to move all the validation checks into a local function and then use the function to validate state after both handler calls **and** between the first and second handler call.

Reply:
Indeed, and it has uncovered a small bug!